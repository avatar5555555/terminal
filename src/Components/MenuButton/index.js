// @flow
import React, { Fragment } from 'react'
import Menu from '@material-ui/core/Menu'
import MenuIcon from '@material-ui/icons/Menu'
import MenuItem from '@material-ui/core/MenuItem'
import IconButton from '@material-ui/core/IconButton'

type Props = {
  handleOpen: Function,
  handleClose: Function,
  anchor: HTMLElement | null
}

const MenuButton = ({ handleOpen, handleClose, anchor }: Props) => (
  <Fragment>
    <IconButton color="inherit" aria-label="Menu" onClick={handleOpen}>
      <MenuIcon />
    </IconButton>
    <Menu
      id="simple-menu"
      anchorEl={anchor}
      open={Boolean(anchor)}
      onClose={handleClose}
    >
      <MenuItem id="menu-item" onClick={handleClose}>
        Logout
      </MenuItem>
    </Menu>
  </Fragment>
)

export default MenuButton
