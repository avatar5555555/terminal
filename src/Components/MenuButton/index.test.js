import React from 'react'
import { shallow } from 'enzyme'

import MenuButton from '.'

describe('MenuButton', () => {
  it('renders', () => {
    const wrapper = shallow(<MenuButton />)

    expect(wrapper).toMatchSnapshot()
  })
})
