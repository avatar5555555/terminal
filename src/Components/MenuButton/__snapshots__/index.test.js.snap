// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuButton renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MenuButton />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <WithStyles(IconButton)
          aria-label="Menu"
          color="inherit"
          onClick={undefined}
>
          <pure(Menu) />
</WithStyles(IconButton)>,
        <WithStyles(Menu)
          anchorEl={undefined}
          id="simple-menu"
          onClose={undefined}
          open={false}
>
          <WithStyles(MenuItem)
                    id="menu-item"
                    onClick={undefined}
          >
                    Logout
          </WithStyles(MenuItem)>
</WithStyles(Menu)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "aria-label": "Menu",
          "children": <pure(Menu) />,
          "color": "inherit",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "anchorEl": undefined,
          "children": <WithStyles(MenuItem)
            id="menu-item"
            onClick={undefined}
>
            Logout
</WithStyles(MenuItem)>,
          "id": "simple-menu",
          "onClose": undefined,
          "open": false,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Logout",
            "id": "menu-item",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "Logout",
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <WithStyles(IconButton)
            aria-label="Menu"
            color="inherit"
            onClick={undefined}
>
            <pure(Menu) />
</WithStyles(IconButton)>,
          <WithStyles(Menu)
            anchorEl={undefined}
            id="simple-menu"
            onClose={undefined}
            open={false}
>
            <WithStyles(MenuItem)
                        id="menu-item"
                        onClick={undefined}
            >
                        Logout
            </WithStyles(MenuItem)>
</WithStyles(Menu)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "aria-label": "Menu",
            "children": <pure(Menu) />,
            "color": "inherit",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "anchorEl": undefined,
            "children": <WithStyles(MenuItem)
              id="menu-item"
              onClick={undefined}
>
              Logout
</WithStyles(MenuItem)>,
            "id": "simple-menu",
            "onClose": undefined,
            "open": false,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Logout",
              "id": "menu-item",
              "onClick": undefined,
            },
            "ref": null,
            "rendered": "Logout",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
